aggregator_core/src/datastore.rs

```diff
@@ -3561,12 +3505,22 @@ impl<C: Clock> Transaction<'_, C> {
                     (SELECT id FROM tasks WHERE task_id = $1), $2, $3, $4, $5, $6, $7, $8, $9, $10,
                     $11, $12, $13, $14, $15
                 )
-                ON CONFLICT DO NOTHING
-                RETURNING COALESCE(GREATEST(UPPER(COALESCE($3::TSRANGE, $6::TSRANGE)), (SELECT MAX(UPPER(COALESCE(batch_interval, client_timestamp_interval))) FROM batch_aggregations ba WHERE ba.task_id = batch_aggregations.task_id AND ba.batch_identifier = batch_aggregations.batch_identifier AND ba.aggregation_param = batch_aggregations.aggregation_param)) < COALESCE($16::TIMESTAMP - (SELECT report_expiry_age FROM tasks WHERE task_id = $1) * '1 second'::INTERVAL, '-infinity'::TIMESTAMP), FALSE) AS is_expired",
+                ON CONFLICT(task_id, batch_identifier, aggregation_param, ord) DO UPDATE
+                    SET (
+                        client_timestamp_interval, state, aggregate_share, report_count, checksum,
+                        aggregation_jobs_created, aggregation_jobs_terminated, created_at,
+                        updated_at, updated_by
+                    ) = (
+                        excluded.client_timestamp_interval, excluded.state,
+                        excluded.aggregate_share, excluded.report_count, excluded.checksum,
+                        excluded.aggregation_jobs_created, excluded.aggregation_jobs_terminated,
+                        excluded.created_at, excluded.updated_at, excluded.updated_by
+                    )
+                    WHERE GREATEST(UPPER(COALESCE(batch_aggregations.batch_interval, batch_aggregations.client_timestamp_interval)), (SELECT MAX(UPPER(COALESCE(ba.batch_interval, ba.client_timestamp_interval))) FROM batch_aggregations ba WHERE ba.task_id = batch_aggregations.task_id AND ba.batch_identifier = batch_aggregations.batch_identifier AND ba.aggregation_param = batch_aggregations.aggregation_param)) < COALESCE($16::TIMESTAMP - (SELECT report_expiry_age FROM tasks WHERE id = batch_aggregations.task_id) * '1 second'::INTERVAL, '-infinity'::TIMESTAMP)",
```

Comment:
Not to rehash older discussions on this, but I do really find this tough to read when it's a single line, to such an extent that I have to break it out in my editor. No changes necessary right now I think, I'll just make a note in https://github.com/divviup/janus/issues/2562 for us to decide on a style and pass through the GC queries.

Reply:
Yeah, this is fair; these lines started pretty long and are now enormous. I think I'm going to address this in the same follow-on PR that takes the approach in #2807 -- another benefit of query-splitting is that it will make these queries somewhat more tractable.